DataFormats/HcalRecHit/interface/HcalSpecialTimes.h

```diff
@@ -50,21 +61,29 @@ namespace HcalSpecialTimes
         constexpr float tdc_to_ns = 0.5f;
 
         constexpr int six_bits_mask = 0x3f;
+        constexpr int tdc_code_largestnormal = 49;
+        constexpr int tdc_code_invalid = 58;
         constexpr int tdc_code_overshoot = 62;
         constexpr int tdc_code_undershoot = 63;
-        constexpr int tdc_code_invalid = 58;
 
         float t = tdc_to_ns*tdc;
         if (tdc > six_bits_mask || tdc < 0)
             t = UNKNOWN_T_INVALID_RANGE;
-        else if (tdc == tdc_code_overshoot)
-            t = UNKNOWN_T_OVERSHOOT;
-        else if (tdc == tdc_code_undershoot)
-            t = UNKNOWN_T_UNDERSHOOT;
-        else if (tdc == tdc_code_invalid)
-            t = UNKNOWN_T_INVALID_CODE;
-        else if (tdc > tdc_code_invalid)
-            t = UNKNOWN_T_DLL_FAILURE;
+        else if (tdc > tdc_code_largestnormal)
+        {
+            // The undershoot code happens by far more often
+            // than any other special code. So check for it first.
+            if (tdc == tdc_code_undershoot)
+                t = UNKNOWN_T_UNDERSHOOT;
+            else if (tdc == tdc_code_overshoot)
+                t = UNKNOWN_T_OVERSHOOT;
+            else if (tdc > tdc_code_invalid)
+                t = UNKNOWN_T_DLL_FAILURE;
+            else if (tdc == tdc_code_invalid)
+                t = UNKNOWN_T_INVALID_CODE;
+            else
+                t = UNKNOWN_T_50TO57;
```

Comment:
please make the choice of the range of values here explicit to make this more readable,
or/and perhaps put the ```else if (tdc > tdc_code_invalid)``` at the end to have a continuous coverage of values.

Reply:
This code is as readable as it can be. It tries all special TDC values from the top down. If you are unhappy about it, you are welcome to write your own.

Reply:
Is there a special reason to cover values out of order?
If not, please reorder to cover values sequentially.

Reply:
They are not out of order. The coverage is actually top down, starting from 63. This is also efficient, as 63 is the most common special code and 62 is the second most common, with everything else in the noise.

Reply:
@igv4321 : of course the logic is correct also in your case. However, a cleaner and more easily readable sequence could be:

```
       else if (tdc > tdc_code_largestnormal)
       {
           // The undershoot code happens by far more often
           // than any other special code. So check for it first.
           if (tdc == tdc_code_undershoot)
               t = UNKNOWN_T_UNDERSHOOT;
           else if (tdc == tdc_code_overshoot)
               t = UNKNOWN_T_OVERSHOOT;
           else if (tdc == tdc_code_invalid)
               t = UNKNOWN_T_INVALID_CODE;
           else if (tdc < tdc_code_invalid)
               t = UNKNOWN_T_50TO57;
           else
               t = UNKNOWN_T_DLL_FAILURE;
       } 
```

The rationale could be the following: the next developer that needs to add another failure code (or range of them, as you are doing in this PR) can easily add the corresponding "if" statement in the correct place with no risk of error. Think for example if you want to consider the codes 59-60-61: here you can simply add the corresponding range before the last "else", while in your formulation you need to rewrite several "if" statements (with the risk of error, as long as the code intervals add up and become more convoluted).

Reply:
This has to be weighted against the rationale of not modifying the code which is already written, tested, and working as expected. I think that the nebulous possibility that somebody else can make a mistake in some particular scenario does not stand a chance. Besides, your suggestion contradicts that of Slava -- why don't you guys fight it out between yourself first.

Reply:
it's the same comment from both of us: to make the range explicit in the conditional expression.

Reply:
Qslava77 Are you happy about @perrotta proposal? I don't care since it does not dumb down my code too much.

Reply:
@igv4321 : no, my suggestion does not contradict what Slava said. While the checks on the individual codes are left at the beginning (as you say that they are the ones which show up more often), the ranges of codes are ordered as requested. If I read the last ```else``` I can immediately understand that it includes whatever has ```tdc > tdc_code_invalid``` and was not considered indiviually in the first lines. With your formulation I had to go through all previous lines to understand what was considered as ```else```.
(Even clearer if you further  make the range explicit also in the last conditional expression)

Reply:
> Qslava77 Are you happy about @perrotta proposal? I don't care since it does not dumb down my code too much.

yes, I have no other suggestions on this part of the changes